Engine/Commands/InvokeScriptAnalyzerCommand.cs

```diff
@@ -379,8 +419,26 @@ private void AnalyzeFile(string filePath)
             {
                 foreach (ICommandRule commandRule in ScriptAnalyzer.Instance.CommandRules)
                 {
-                    if ((includeRule == null || includeRule.Contains(commandRule.GetName(), StringComparer.OrdinalIgnoreCase)) &&
-                        (excludeRule == null || !excludeRule.Contains(commandRule.GetName(), StringComparer.OrdinalIgnoreCase)))
+                    bool includeRegexMatch = false;
+                    bool excludeRegexMatch = false;
+                    foreach (Regex include in includeRegexList)
+                    {
+                        if (include.IsMatch(commandRule.GetName()))
+                        {
+                            includeRegexMatch = true;
+                            break;
+                        }
+                    }
+                    foreach (Regex exclude in excludeRegexList)
+                    {
+                        if (exclude.IsMatch(commandRule.GetName()))
+                        {
+                            excludeRegexMatch = true;
+                            break;
+                        }
+                    }
+                    if ((includeRule == null || includeRule.Contains(commandRule.GetName(), StringComparer.OrdinalIgnoreCase) || includeRegexMatch) &&
```

Comment:
Do we still need this check anymore? Since this should already be done in the regex code above.